libraries/provider_firewall_nftables.rb

```diff
@@ -0,0 +1,157 @@
+#
+# Author:: Matthias Pausch (<>)
+# Cookbook:: firewall
+# Resource:: default
+#
+# Copyright:: 2011-2019, Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+class Chef
```

Comment:
If we're adding a new resource, we should be adding it via the custom resource pattern rather than the old LWRP way.

Reply:
Ok, I did something, but I am not sure if I did it correctly.  So basically I copied `libraries/provider_firewall_nftables.rb` over to `resource/firewall_nftables.rb` and added some properties, that are already defined as attributes in `libraries/resource_firewall.rb`.